src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -680,22 +651,22 @@ public ReadOnlyTensorSpan<T> Slice(params scoped ReadOnlySpan<NRange> ranges)
             if (ranges.Length != Lengths.Length)
                 throw new ArgumentOutOfRangeException(nameof(ranges), "Number of dimensions to slice does not equal the number of dimensions in the span");
 
-            scoped Span<nint> shape;
+            scoped Span<nint> lengths;
             scoped Span<nint> offsets;
-            if (Rank > 6)
+            if (Rank > TensorSpan.MaxRankForStackAlloc)
             {
-                shape = stackalloc nint[Rank];
+                lengths = stackalloc nint[Rank];
                 offsets = stackalloc nint[Rank];
```

Comment:
nit: These should be using `TensorSpan.MaxRankForStackAlloc` and then slicing to the actual rank. Doing a stackalloc of unknown size is typically bad.